apache2/re.c

```diff
@@ -1970,6 +1970,48 @@ msre_ruleset *msre_ruleset_create(msre_engine *engine, apr_pool_t *mp) {
 
     return ruleset;
 }
+/**
+
+
+ * Creates a ruleset by copy the parent ruleset.
+ *
+ * @param mp apr pool structure
+ * @param parent_ruleset Parent's msre_ruleset
+ *
+ * return the address of created ruleset if OK.
+ * return NULL if Something went wrong.
+ */
+
+msre_ruleset *msre_ruleset_create_and_copy(msre_engine *engine, apr_pool_t *mp, msre_ruleset *parent_ruleset) {
+    msre_ruleset *child_ruleset;
+
+    child_ruleset = apr_pcalloc(mp, sizeof(msre_ruleset));
+    if (child_ruleset == NULL) return NULL;
+    child_ruleset->mp = mp;
+    child_ruleset->engine = engine;
+
+    if (parent_ruleset == NULL) {
+        child_ruleset->phase_request_headers  = apr_array_make(mp, 25, sizeof(const msre_rule *));
```

Comment:
why was 25 chosen? also please use define or variable to store this value

Reply:
Because we want to merge the msre_ruleset_create and copy_rules into one function. 
(1) About apr_array_make(): In the original msre_ruleset_create() function, it use apr_array_make(ruleset->mp, 25, ..) to init the child ruleset, which doesn't use  define or variable to store this value. If we implement in this way, we cannot keep the same coding style as previous version. Do you have some other suggestions on it? 
(2) About child: Because the original code create the child ruleset and allocate memory first, we have to allocate memory for the child in order to keep our logic is the same as the original code.

Reply:
i understand the reason of keeping the prev code style, however once you alloc in two different place there might be a chance for future changes that will miss one of those place and there will be a miss match (one will allocate 25 while the other will allocate 30) - however this is minor so feel free to ignore it.